studio/components/layouts/ProjectLayout/PausingState.tsx

```diff
@@ -1,11 +1,42 @@
-import { Project } from 'types'
+import { useQueryClient } from '@tanstack/react-query'
+import { useParams } from 'common'
 import { Badge, IconCircle, IconLoader } from 'ui'
 
+import { useProjectStatusQuery } from 'data/projects/project-status-query'
+import { invalidateProjectsQuery } from 'data/projects/projects-query'
+import { PROJECT_STATUS } from 'lib/constants'
+import { timeout } from 'lib/helpers'
+import { Project } from 'types'
+import { useEffect, useState } from 'react'
+
 export interface PausingStateProps {
   project: Project
 }
 
 const PausingState = ({ project }: PausingStateProps) => {
+  const { ref } = useParams()
+  const queryClient = useQueryClient()
+  const [startPolling, setStartPolling] = useState(false)
+
+  const { refetch } = useProjectStatusQuery(
+    { projectRef: ref },
+    {
+      enabled: startPolling,
+      onSuccess: async (res) => {
+        if ([PROJECT_STATUS.GOING_DOWN, PROJECT_STATUS.PAUSING].includes(res.status)) {
+          await timeout(2000)
+          refetch()
+        } else if (PROJECT_STATUS.INACTIVE) {
+          await invalidateProjectsQuery(queryClient)
+        }
```

Comment:
I'm still getting stuck on the pausing state.

react-query has built-in support for refetching automatically btw: look for `refetchInterval` in the docs.

It can also be a function which returns a number (interval ms) or `false` to disable the refetching, so you can move this logic into that :)

Reply:
gotcha - hmm is the request not getting triggered when you hit pause? the only problem i can think of is that the async process to toggle a project to pausing takes longer than the initial 4 seconds before the first call to project's status

Reply:
Yeah the pause seems to work, but then I don't see any requests for the polling

Reply:
oh wait lol sorry! the else if logic is wrong 🤦